[quant][be] Rewrite xnnpack_quantizer_utils.py to use decorators

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108920

Summary:
att

Test Plan:
python test/test_quantization.py TestQuantizePT2E

Reviewers:

Subscribers:

Tasks:

Tags:

jerryzh168 committed:
[quant][be] Rewrite xnnpack_quantizer_utils.py to use decorators

Summary:
att

Test Plan:
python test/test_quantization.py TestQuantizePT2E

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel commented on torch/ao/quantization/quantizer/xnnpack_quantizer_utils.py:
```diff
@@ -52,6 +52,23 @@ class QuantizationConfig:
+
+def register_annotator(op: str):
+    def decorator(annotator: AnnotatorType):
+        OP_TO_ANNOTATOR[op] = annotator
```
nit: I think it would make sense to add `annotate` function here that looks up the string in the dict.

jerryzh168 replied:
sorry what do you mean? could you write a small example?

kimishpatel replied:
sorry, misplaced comments. What i meant was how it is used in xnnpack_quantizer.

```
def annotate(op: str, qconfig,...):
  return OP_TO_ANNOTATOR[op](qconfig,....)

from xnnpack_quantizer_utils import annotate

....
def _annotate_conv2d_relu(..)
   annotate("conv2d_relu",...)
```

jerryzh168 replied:
oh OK I can refactor it in next PR

jerryzh168 commented:
@pytorchbot merge

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 committed:
Update on "[quant][be] Rewrite xnnpack_quantizer_utils.py to use decorators"


Summary:
att

Test Plan:
python test/test_quantization.py TestQuantizePT2E

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@pytorchbot merge
